### PR TITLE
scheduling local notifications

### PR DESCRIPTION
**Pull request by renzosal**
This PR contains initial changes for scheduling local notifications based on https://github.com/Neson/react-native-system-notification  and https://github.com/zo0r/react-native-push-notification.

Thanks @RakanNimer for your initial notification PR 

adding priority property for android when presenting a notifications.

Also includes repeatEvery property for both android and IOS.
this is coming to PushNotificationIOS https://github.com/facebook/react-native/pull/7219 so this might change in the future.

WIP

### CODE REVIEW

File: ios/RNFIRMesssaging.m [hunk 1]
```diff
@@ -17,6 +17,29 @@
...
+  notification.fireDate = [RCTConvert NSDate:details[@"fireDate"]] ?: [NSDate date];
+  notification.alertBody = [RCTConvert NSString:details[@"alertBody"]];
+  notification.alertAction = [RCTConvert NSString:details[@"alertAction"]];
+   notification.soundName = [RCTConvert NSString:details[@"soundName"]] ?: UILocalNotificationDefaultSoundName;
+   notification.userInfo = [RCTConvert NSDictionary:details[@"userInfo"]];
+   notification.category = [RCTConvert NSString:details[@"category"]];
+
+  NSNumber *repeatInterval = details[@"repeatInterval"];
+
+  if (repeatInterval) {
```
Comment (evollu) [inline ios/RNFIRMesssaging.m]: why does this need if check while  others don't? because it is integer?

File: index.js [hunk 2]
```diff
@@ -1,10 +1,19 @@
-import {NativeModules, DeviceEventEmitter} from 'react-native';
+import {NativeModules, DeviceEventEmitter, Platform, PushNotificationIOS} from 'react-native';
```
Comment (evollu) [inline index.js]: do we even need to include PushNotificationIOS? most of the functions are implemented

File: index.js [hunk 3]
```diff
@@ -17,13 +26,65 @@ FCM.requestPermissions = () => {
...
+FCM.scheduleLocalNotification = function(details) {
+    var iosNotification = {
+        fireDate: details.date.getTime(),
+        alertBody: details.message,
+        userInfo: details.userInfo,
+        repeatInterval: REPEAT_INTERVAL_IOS[details.repeatEvery] || 0
+    };
+    details.fireDate = details.date.getTime();
+    delete details.date;
+    FIRMessaging.scheduleLocalNotification((Platform.OS === 'ios')? iosNotification : details);
```
Comment (evollu) [inline index.js]: what is the difference between iosNotification vs detail?

File: index.js (hunk 2 above)
Comment (evollu) [inline index.js]: since we include PushNotificationIOS here, can we bridge request permission and check permission on it?

File: index.js [hunk 4]
```diff
@@ -17,13 +26,65 @@ FCM.requestPermissions = () => {
...
+    }
+};
+
+FCM.cancelAll = () => {
+    if (Platform.OS ==='android'){
+        FIRMessaging.cancelAll();
+    }
+    else if (Platform.OS ==='ios') {
+        PushNotificationIOS.cancelLocalNotifications();
+        PushNotificationIOS.setApplicationIconBadgeNumber(0);
```
Comment (evollu) [inline index.js]: cancel local notification shouldn't clear badges. there could be outstanding remote notification

File: index.js [hunk 5]
```diff
@@ -17,13 +26,65 @@ FCM.requestPermissions = () => {
...
+
+FCM.cancel = (notificationID) => {
+    if(Platform.OS === 'android'){
+        FIRMessaging.cancel(notificationID);
+    }
+};
+
+FCM.cancelAll = () => {
+    if (Platform.OS ==='android'){
+        FIRMessaging.cancelAll();
```
Comment (evollu) [inline index.js]: can we name this as `cancelLocalNotifications` to be consistent with ios version?

File: index.js [hunk 6]
```diff
@@ -17,13 +26,65 @@ FCM.requestPermissions = () => {
...
+        userInfo: details.userInfo,
+        repeatInterval: REPEAT_INTERVAL_IOS[details.repeatEvery] || 0
+    };
+    details.fireDate = details.date.getTime();
+    delete details.date;
+    FIRMessaging.scheduleLocalNotification((Platform.OS === 'ios')? iosNotification : details);
+};
+
+FCM.cancel = (notificationID) => {
+    if(Platform.OS === 'android'){
```
Comment (evollu) [inline index.js]: ios doesn't have this function?

File: index.js [hunk 7]
```diff
@@ -17,13 +26,65 @@ FCM.requestPermissions = () => {
...
+    else if (Platform.OS ==='ios') {
+        PushNotificationIOS.cancelLocalNotifications();
+        PushNotificationIOS.setApplicationIconBadgeNumber(0);
+    }
+};
+
 FCM.on = (event, callback) => {
     const nativeEvent = eventsMap[event];
 
+    if(Platform.OS === 'ios'){
```
Comment (evollu) [inline index.js]: can you create a new event type like `onLocalMessageReceived`?

File: index.js (hunk 6 above)
Comment (renzosal) [inline index.js]: nope unfortunately there is no Id on ios, it is based on userInfo

File: index.js (hunk 3 above)
Comment (renzosal) [inline index.js]: iosNotification is just the object with the properties that are specific for PushNotificationIOS

**Pull request closed without merging**